packages/suite/src/components/suite/DeviceDisplay.tsx

```diff
@@ -0,0 +1,211 @@
+        chunks?.map((row, rowIndex) => (
+            <Row
+                // eslint-disable-next-line react/no-array-index-key
+                key={rowIndex}
+                isAlignedRight={rowIndex === 0 && isNextAddress}
+            >
+                {row.map((chunk, chunkIndex) => (
+                    // eslint-disable-next-line react/no-array-index-key
+                    <Text isPixelType={isPixelType} key={chunkIndex}>
+                        {chunk}
+                    </Text>
+                ))}
+            </Row>
+        ));
+
+    const renderChunks = (address: string) => {
+        const chunks: Array<string | ReactElement> | null = address.match(/.{1,4}/g);
+
+        if (isCardano) {
+            // Add pagination icons
+            chunks?.splice(15, 0, <Icon {...iconConfig} icon={iconNextName} />);
+            chunks?.splice(16, 0, <Icon {...iconConfig} icon={iconContinuesName} />);
+        }
+
+        const groupedChunks = chunks?.reduce((result, chunk, index) => {
+            const rowIndex = Math.floor(index / 4);
+
+            if (!result[rowIndex]) {
+                result[rowIndex] = [];
+            }
+
+            result[rowIndex].push(chunk);
+
+            return result;
+        }, [] as ReactNode[][]);
+
+        const handleOnCopy = (event: ClipboardEvent) => {
+            const selectedText = window.getSelection()?.toString();
+
+            if (selectedText) {
+                const processedText = selectedText.replace(/\s/g, '');
+                event?.nativeEvent?.clipboardData?.setData('text/plain', processedText);
+                event.preventDefault();
+            }
+        };
+
+        // this is now not possible but will be used once Cardano flow is improved
+        if (isCardano) {
+            const PAGE_SIZE = 4;
+            return (
+                <>
+                    <Chunks onCopy={handleOnCopy}>
+                        {showChunksInRows(groupedChunks?.slice(0, PAGE_SIZE))}
+                    </Chunks>
+                    <Wrapper>
+                        <Divider areChunksUsed={areChunksUsed} />
+                        <AddressLabel areChunksUsed={areChunksUsed}>
+                            <Translation id="NEXT_PAGE" />
+                        </AddressLabel>
+                    </Wrapper>
+                    <Chunks onCopy={handleOnCopy}>
+                        {showChunksInRows(groupedChunks?.slice(PAGE_SIZE), true)}
+                    </Chunks>
+                </>
+            );
+        }
+
+        return <Chunks onCopy={handleOnCopy}>{showChunksInRows(groupedChunks)}</Chunks>;
+    };
+
+    const renderOriginal = (address: string) => {
+        if (isCardano) {
+            const breakpoint = isPixelType ? 70 : 81;
+            return (
+                <>
+                    <Text isPixelType={isPixelType}>{address.slice(0, breakpoint)}</Text>
+                    <StyledNextIcon {...iconConfig} isPixelType={isPixelType} icon={iconNextName} />
```

Comment:
nit: to avoid relative positioning and be more flexible with address lengths, could do this on the Icon:

```
    display: inline-flex;
    margin-left: 12px;
```

Reply:
Tried it but it doesn't work much
![BB8FDFF8-FCA7-4489-864C-CA1D416A7C02_4_5005_c](https://github.com/trezor/trezor-suite/assets/66002635/7d2c902b-81d3-4af5-b12e-eab4382eca78)

Reply:
[This](https://github.com/trezor/trezor-suite/pull/9671/commits/a45956031004d793af8672892ec44a3e4883a9c8) is what I meant :)